### ISSUE TITLE
How to fine-tune on my own datasete for cloud completion?

### ISSUE THREAD

**Issue by FSet89** [JohnRomanelis/SPVD#14]
I have a very small dataset with simple point clouds. How should I use this code to perform finetuning on a pretrained model?

**Comment by JohnRomanelis** [comment]
Finetuning on smaller datasets was not something we explored within the scope of this work, but here are two approaches you could try:

1. **Retrain from scratch with your class added:** Add your own class to the ShapeNet dataset and train the class-conditioned model from the beginning. If you look at the training code this should be fairly straightforward to set up, but depending on your hardware this could take a day or two to complete.

2. **Finetune the pretrained model:**  Add a new class embedding for your data and continue training from the pretrained checkpoint of the class conditioned model. You can train this only with your own class or mix in some ShapeNet data to avoid degrading the existing weights. Make sure to use a small learning rate to preserve the pretrained representations. This approach requires a bit more code changes but should need significantly less training time.

Feel free to reach out if you need further assistance setting either of these up!
